server/system/service/smtp_configuration_checker.go

```diff
@@ -0,0 +1,215 @@
+package service
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/cortezaproject/corteza/server/pkg/actionlog"
+	intAuth "github.com/cortezaproject/corteza/server/pkg/auth"
+	"github.com/cortezaproject/corteza/server/pkg/mail"
+	"github.com/cortezaproject/corteza/server/pkg/options"
+	"github.com/cortezaproject/corteza/server/system/types"
+	gomail "gopkg.in/mail.v2"
+	htpl "html/template"
+	"io/ioutil"
+	"regexp"
+	"strings"
+)
+
+type (
+	smtpConfigurationChecker struct {
+		actionlog actionlog.Recorder
+		settings  *types.AppSettings
+		ts        TemplateService
+		opt       options.AuthOpt
+	}
+)
+
+const (
+	hostCheckRE = "^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])$"
+)
+
+func SmtpConfigurationChecker(s *types.AppSettings, ts TemplateService, opt options.AuthOpt) *smtpConfigurationChecker {
+	return &smtpConfigurationChecker{
+		actionlog: DefaultActionlog,
+		settings:  s,
+		ts:        ts,
+		opt:       opt,
+	}
+}
+
+func (svc smtpConfigurationChecker) SmtpConfigurationCheck(ctx context.Context, smtpConfigs *types.SmtpConfiguration) (checkResults *types.SmtpCheckResult, err error) {
```

Comment:
- We talked about this - rename it to `Check`
 - What I'm also missing here is any kind of access control; anyone can call this function and we do not want that. Check if the user can manage settings (`CanManageSettings`).